CMakeLists.txt

```diff
@@ -31,33 +31,46 @@ elseif (WIN32)
 endif ()
 
 # Will be overwritten during the cmake initialization.
+
+# Value can be one of "None" and "ControlFlow"
+# If None, no LVI mitigation will be applied
+# If ControlFlow, only control-flow mitigations will be applied
+# Support of All LVI hardening (-mlvi-hardening) is omitted
 set(LVI_MITIGATION
     "None"
-    CACHE STRING "Build with LVI mitigation.")
+    CACHE STRING "Build with LVI mitigation. Options: [None, ControlFlow]")
 set(LVI_MITIGATION_BINDIR
     "None"
-    CACHE STRING "Path to the LVI mitigation bindir.")
+    CACHE STRING "[clang-10 only] Path to the LVI mitigation bindir.")
 
 # On Unix, configuring cmake to use the preferred compiler.
 # Note that this must be done before the `project` command.
 if (UNIX)
-  if (LVI_MITIGATION MATCHES ControlFlow)
-    # If the LVI mitigation is enabled, use the customized compilation toolchain.
+  # For the normal build, if the CC environment variable has been specified or if
+  # the CMAKE_C_COMPILER cmake variable has been passed to cmake, use the C compiler
+  # that has been specified. Otherwise, prefer clang. Same for C++ compiler.
+  if (NOT DEFINED ENV{CC} AND NOT DEFINED CMAKE_C_COMPILER)
+    find_program(CMAKE_C_COMPILER clang-11 clang-10 clang)
+  endif ()
+  if (NOT DEFINED ENV{CXX} AND NOT DEFINED CMAKE_CXX_COMPILER)
+    find_program(CMAKE_CXX_COMPILER clang++-11 clang++-10 clang++)
+  endif ()
+
+  if (CMAKE_C_COMPILER MATCHES "10$" OR CMAKE_CXX_COMPILER MATCHES "10$")
```

Comment:
This change is good, but it seems worthwhile to point out that `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPLER` ends with `clang.exe` rather than `clang-10` and thus `OE_USING_CLANG_10` won't be defined on Windows. 

It would be less confusing to have `OE_USING_CLANG_10` set according to whether clang 10 is used regardless of whether it is building on UNIX or Windows. I have some work in progress that I can adapt to address this. A preview of it is: 
https://github.com/openenclave/openenclave/commit/c29e96985cca67471eae3708f4453c8e47e3d562

Alternatively, we may want to consider logic such as `CMAKE_C_COMPILER_VERSION less than 10` or `CMAKE_C_COMPILER_VERSION greater than 10.99` so that we don't have to implement logic specific to a single version.

Reply:
Yes, for the most part it is fine. The only part is that Windows may have issues as there are checks for `OE_USING_CLANG_10` while building in Windows, such as the case in enclave/core/CMakeLists.txt in this PR